test/unit/range/container/bitcompressed_vector_test.cpp

```diff
@@ -7,6 +7,35 @@
 
 #include <gtest/gtest.h>
 
+#include <seqan3/alphabet/nucleotide/concept.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/views/complement.hpp>
+
 #include "container_test_template.hpp"
 
 INSTANTIATE_TYPED_TEST_SUITE_P(bitcompressed, container_over_dna4_test, seqan3::bitcompressed_vector<seqan3::dna4>, );
+
+TEST(bitcompressed_vector_test, issue1743_complement_on_proxy)
+{ // https://github.com/seqan/seqan3/issues/1743
+
+    seqan3::bitcompressed_vector<seqan3::dna4> v{seqan3::dna4{}, seqan3::dna4{}};
+
+    auto proxy = *v.begin();
+    auto complement = seqan3::complement(proxy);
+
+    EXPECT_TRUE((std::same_as<decltype(complement), seqan3::dna4>));
+    EXPECT_EQ(complement, seqan3::assign_rank_to(3, seqan3::dna4{}));
+}
+
+TEST(bitcompressed_vector_test, issue1743_view_combinability)
+{ // https://github.com/seqan/seqan3/issues/1743
+
+    seqan3::bitcompressed_vector<seqan3::dna4> v{seqan3::dna4{}, seqan3::dna4{}, seqan3::dna4{}, seqan3::dna4{}};
+
+    auto complement = v | seqan3::views::complement;
+
+    EXPECT_EQ(v.size(), complement.size());
+
+    for (auto chr : complement)
+        EXPECT_EQ(chr, seqan3::assign_rank_to(3, seqan3::dna4{}));
```

Comment:
```suggestion
    RANGE_EXPECT_EQ(complement, seqan3::dna4_vector{'T'_dna4, 'G'_dna4, 'C'_dna4, 'A'_dna4});
```
And make the vector above:

```cpp
seqan3::dna4_vector{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4}
```